src/base/rss/rssdownloadrule.cpp

```diff
@@ -42,33 +42,43 @@
 
 using namespace Rss;
 
-const QString episodeRegex = "[^A-Za-z0-9]((?:S\\d{1,}(?:E\\d{1,})+)|(?:\\d{1,}x\\d{1,})|(?:\\d{4}\\.\\d{2}\\.\\d{2}))[^A-Za-z0-9]";
+static const QString episodeRegex = "(?:^|[^A-Za-z0-9])((?:S\\d{1,}(?:E\\d{1,})+)|(?:\\d{1,}x\\d{1,})|(?:\\d{4}\\.\\d{2}\\.\\d{2}))(?:[^A-Za-z0-9]|$)";
 
-QString computeEpisodeName(QString matchedEpisode)
+QString computeEpisodeName(const QString &article)
 {
+    QRegExp f(episodeRegex, Qt::CaseInsensitive);
+    // See if we can extract an season/episode number or date from the title
+    if (f.indexIn(article) == -1) {
+        return QString();
+    }
+
+    // Extract out the episode identifier
+    QString matchedEpisode = f.cap(1);
+
+    // Is it in the S01E01 format?
     if (matchedEpisode.startsWith('S')) {
```

Comment:
Though the regex matching is case insensitive, this test will erroneously fail if the episode number in the article name is using lower case letters, e.g. `s01e01`.